src/static/js/pluginfw/plugins.js

```diff
@@ -143,15 +143,144 @@ async function loadPlugin(packages, plugin_name, plugins, parts) {
+    var conflictObj = {};
+    var conflictsPre = conflictMatcher(parts, name, 'pre');
+    var conflictsPost = conflictMatcher(parts, name, 'post');
+
+    conflictObj[name] = {pre: conflictsPre, post: conflictsPost};
+    conflicts.push(conflictObj);
+  });
+
+  return conflicts;
+}
+
+function closedChainChecker (parts, name, match, matchCount, res) {
+  var noChain = true;
+  for (var i = 0; i < res.length; i++ ) {
+      var item = res[i];
+
+      if (item[0] === name || item[1] === name) {
+        if (item[0] === match || item[1] === match) {
+          //value allready exists
+          noChain = false;
+            // check which plugins has stricter pre/post definition
+          _.each(parts[match].pre || [], function (item_name) {
+            var matches = matcher(item_name, _.chain(parts).keys().value());
+            if (matches.length > matchCount) {
+              res.splice(i, 1);
+              noChain = true;
+            }
+          });
+          _.each(parts[match].post || [], function (item_name) {
+            var matches = matcher(item_name, _.chain(parts).keys().value());
+            if (matches.length > matchCount) {
+              res.splice(i, 1);
+              noChain = true;
+            }
+          });
+
+          i = res.length;
+          break;
+        }
+      }
+  }
+
+  return noChain;
+}
+
+function matchHooks (parts, name, res, conflicts, type) {
+  var names = _.chain(parts).keys().value();
+  _.each(parts[name][type], function (item_name)  {
+    var matchedNames = matcher(item_name, names);
+    _.each(matchedNames, function (matchName) {
+      if (name !== matchName) {
+        var noConflict = true;
+
+        if (conflicts) {
+          _.each(conflicts, function (conflictObject) {
+            if (conflictObject[item_name] && (conflictObject[item_name].matches && conflictObject[item_name].matches.indexOf(matchName) > -1)) {
+              noConflict = false;
+            }
+          })
+        }
+        if (noConflict) {
+          noConflict = closedChainChecker(parts, name, matchName, matchedNames.length, res);
+        }
+
+        if (noConflict && type === 'post') {
+          res.push([name, matchName]);
+        } else if (noConflict){
+          res.push([matchName, name]);
+        }
+      }
+    });
+  });
+}
+
 function partsToParentChildList(parts) {
   var res = [];
+  var conflicts = checkPluginConflicts(parts);
+
   _.chain(parts).keys().forEach(function (name) {
-    _.each(parts[name].post || [], function (child_name)  {
-      res.push([name, child_name]);
-    });
-    _.each(parts[name].pre || [], function (parent_name)  {
-      res.push([parent_name, name]);
-    });
+    var conflictsPre = [];
+    var conflictsPost = [];
+
+    for (var i=0; i < conflicts.length; i++) {
+      if (conflicts[i][name]) {
+        conflictsPre = conflicts[i][name].pre;
+        conflictsPost = conflicts[i][name].post;
+        i = conflicts.length;
+        break;
```

Comment:
`i = conflicts.length;` comes right before the `break`, and is not used afterwards.